solver/src/solver/http_solver/model.rs

```diff
@@ -57,20 +58,20 @@ pub struct TokenInfoModel {
 #[derive(Debug, Serialize)]
 pub struct CostModel {
     pub amount: u128,
-    pub token: String,
+    pub token: H160,
 }
 
 #[derive(Debug, Serialize)]
 pub struct FeeModel {
     pub amount: u128,
```

Comment:
We can run into precision issues with this as it will encode the amount as a number which most parsers interpret as a IEEE double precision floating point number which would only be accurate to `2^53`. This should probably use `display_fromstr` formatting.